models/Asset.php

```diff
@@ -1555,53 +1555,39 @@ public function getMetadata($name = null, $language = null, $strictMatch = false
             return $transformedData;
         };
 
-        if ($name) {
-            if ($language === null) {
-                $language = Pimcore::getContainer()->get(LocaleServiceInterface::class)->findLocale();
-            }
-
-            $data = null;
-            foreach ($this->metadata as $md) {
-                if ($md['name'] == $name) {
-                    if ($language == $md['language']) {
-                        if ($raw) {
-                            return $md;
-                        }
-
-                        return $convert($md);
-                    }
-                    if (empty($md['language']) && !$strictMatch) {
-                        if ($raw) {
-                            return $md;
-                        }
-                        $data = $md;
-                    }
-                }
-            }
-
-            if ($data) {
-                if ($raw) {
-                    return $data;
-                }
-
-                return $convert($data);
-            }
-
-            return null;
+        if ($language === null) {
+            $language = Pimcore::getContainer()->get(LocaleServiceInterface::class)->findLocale();
         }
 
         $metaData = $this->getObjectVar('metadata');
+
         $result = [];
+
         if (is_array($metaData)) {
             foreach ($metaData as $md) {
                 $md = (array)$md;
-                if (!$raw) {
-                    $md['data'] = $convert($md);
+
+                if($name) {
+                    if ($md['name'] == $name) {
+                        if ($language == $md['language'] || (empty($md['language']) && !$strictMatch)) {
+                            if ($raw) {
+                                return $md;
+                            }
+                            return $convert($md);
+                        }
+                    }
+                } else if ($language == $md['language'] || (empty($md['language']) && !$strictMatch)) {
```

Comment:
The logic is changed and wrong here:
When specific `$language` and `$name` are not set, it should return all the metadata in an array, but with this PR `$language` is now always set at line 1559* and as soon as encounters a metadata row that has a different language (eg. `$language = 'en'` and `$md['language'] = 'de'` not empty), it doesn't match the conditions of this `else if` and exits by returning `null` discarding every other metadata rows stored in `$result[]`